src/plugins/map/MQ2MapCommands.cpp

```diff
@@ -583,30 +595,25 @@ void MapSetLocationCmd(SPAWNINFO* pChar, char* szLine)
 	MapObjectMapLoc* origLoc = GetMapLocByTag(tag);
 	if (origLoc)
 	{
-		params = origLoc->GetParams();
+		params = origLoc->GetParamsClone();
```

Comment:
You don't need to have a function that returns a copy. you can just use straight value assignment.